tests/test_shadow_fork.py

```
@@ -77,26 +77,41 @@ def test_create_fails_if_base_branch_missing(self, tmp_path):
+        import logging
+        caplog.set_level(logging.WARNING, logger="swarm.runtime.shadow_fork")
         fork = ShadowFork(repo_root=tmp_path)
 
         with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, " M file.txt", ""),  # Uncommitted changes exist
-                (True, "", ""),  # Verify base branch exists
-                (True, "", ""),  # Create and switch to shadow branch
-            ]
+            def side_effect_fn(cmd, **kwargs):
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `side_effect_fn`

(ANN202)